src/symbols_linux.cpp

```diff
@@ -804,14 +804,27 @@ void Symbols::parseLibraries(CodeCacheArray* array, bool kernel_symbols) {
 
             // Protect library from unloading while parsing in-memory ELF program headers.
             // Also, dlopen() ensures the library is fully loaded.
-            // Main executable and ld-linux interpreter cannot be dlopen'ed, but dlerror() returns NULL for them.
             void* handle = dlopen(lib.file, RTLD_LAZY | RTLD_NOLOAD);
-            if (handle != NULL || dlerror() == NULL) {
-                ElfParser::parseProgramHeaders(cc, lib.image_base, lib.map_end, OS::isMusl());
-                if (handle != NULL) {
-                    dlclose(handle);
+            if (handle == NULL) {
+                char *dlerror_output = dlerror();
+                // Main executable and ld-linux interpreter cannot be dlopen'ed, but dlerror()
+                // returns NULL for them.
+                if (dlerror_output != NULL) {
+                    char buffer[1024];
+                    // https://git.musl-libc.org/cgit/musl/tree/ldso/dynlink.c?h=v1.2.5#n2167
+                    sprintf(buffer, "Library %s is not already loaded", lib.file);
+                    if (strcmp(dlerror_output, buffer) != 0) {
```

Comment:
I would very much prefer to check an integer error code here, but I couldn't find anything like that

Reply:
This looks a bit ugly. For which libraries is this error returned?
Can we just ignore any dlopen errors on musl, since musl does not support library unloading?

Reply:
This is for the main executable, `dlerror()` returns `Library ... is not already loaded` instead of `NULL`.
> Can we just ignore any dlopen errors on musl, since musl does not support library unloading?
Sure

Reply:
383c29dd221f37c08bcb2258aada68fea5a25108